src/ims-oauth_server_to_server.js

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const aioLogger = require('@adobe/aio-lib-core-logging')('@adobe/aio-lib-ims-oauth:ims-oauth_server_to_server', { provider: 'debug' })
+const { codes: errors } = require('./errors')
+
+/**
+ * Checks whether the configuration data is missing any required keys.
+ *
+ * @private
+ * @param {object} configData the confiuration data to check
+ * @returns {Array} an array of missing keys, if any
+ */
+function configMissingKeys (configData) {
+  aioLogger.debug(`configMissingKeys configData: ${JSON.stringify(configData)}`)
+
+  const missingKeys = []
+  const requiredKeys = [
+    'client_id',
+    'client_secrets',
+    'technical_account_email',
+    'technical_account_id',
```

Comment:
even though we don't use this to get the token, they are required because of the way IMS plugins work: these keys act as a signature for the plugin -- this is how we select what plugin to invoke for a set of credentials.